src/dmd/root/array.d

```diff
@@ -1013,7 +1038,7 @@ enum isInputRange(R) =
 
 /// Evaluates to `true` if `func` can be called with a value of `T` and returns `bool`.
 enum isPredicateOf(alias func, T) =
-    is(ReturnType!(typeof((T t) => func(t))) == bool);
+    __traits(compiles, { auto _ = (T t) { if (func(t)) {} }; });
```

Comment:
```suggestion
    is(typeof((T t) => func(t) ? 0 : 0)));
```

Reply:
or maybe:
```suggestion
    is(typeof(!func(T.init)));
```

Reply:
@BorisCarvajal I did that before. Apparently it doesn't support `ref` delegates: https://github.com/dlang/dmd/pull/11532#discussion_r467517734.

Reply:
I just tested that works.
https://run.dlang.io/is/skNYh7

Reply:
Neither of those delegates accept the parameter by ref ...

Reply:
>parameter by ref

Thanks for pointing out that.

Reply:
Fixed.